repository/kafka/docs/latest/kudo-kafka-runbook.md

```diff
@@ -0,0 +1,128 @@
+
+### Deploying KUDO Kafka with dummy user workload
+
+KUDO Kafka can be deployed with some dummy user workload running over Kafka brokers by specifying `RUN_USER_WORKLOAD` parameter as true. 
+
+In order to install KUDO Kafka having user workload in the namespace `kudo-kafka`:
+```
+kubectl kudo install kafka --instance=kafka --namespace=kudo-kafka \
+ -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
+ -p BROKER_CPUS=200m \
+ -p BROKER_COUNT=3 \
+ -p BROKER_MEM=800m \
+ -p DISK_SIZE=10Gi \
+ -p MIN_INSYNC_REPLICAS=1 \
+ -p NUM_IO_THREADS=10 \
+ -p NUM_NETWORK_THREADS=5 \
+ -p RUN_USER_WORKLOAD=true \
+ -p ENABLE_METRICS=true \
+ -p ADD_SERVICE_MONITOR=true
+ ```
+
+ Here, some parameters which are specified are described as:
+  - `ZOOKEEPER_URI`: Host and Port information for Zookeeper connection.
+  - `BROKER_CPUS`: CPUs (limit) for the Kafka Broker pods.
+  - `BROKER_COUNT`: Number of brokers(pods) spun up for Kafka.
+  - `BROKER_MEM`: Memory (request) for the Kafka Broker pods.
+  - `DISK_SIZE`: Disk size for the brokers.
+  - `MIN_INSYNC_REPLICAS`: When a producer sets acks to "all" (or "-1") MIN_INSYNC_REPLICAS is the minimum number of replicas that must acknowledge a write for the write to be considered successful.
+  - `NUM_IO_THREADS`: Number of threads that the server uses for processing requests.
+  - `NUM_NETWORK_THREADS`: Number of threads that the server uses for receiving requests.
+  - `RUN_USER_WORKLOAD`: Creates some dummy user workload over Kafka brokers.
+  - `ENABLE_METRICS`: Enables metrics endpoint of Kafka to broadcast service metrics over the specified endpoint port.
+  - `ADD_SERVICE_MONITOR`: Creates a service monitor for Kafka service.
+
+
+The output of this command should be something like:
+```
+operator.kudo.dev/v1beta1/kafka created
+operatorversion.kudo.dev/v1beta1/kafka-1.0.1 created
+instance.kudo.dev/v1beta1/kafka created
+```
+
+Other than the desired number of three broker pod tasks, this deployment creates two additional pods i.e. producer and consumer that run over brokers to create some dummy workload.
+We can check these pod tasks running through this:
+```
+$ kubectl get pods -w -n kudo-kafka
+NAME                                            READY   STATUS    RESTARTS   AGE
+kafka-kafka-0                                   1/1     Running   0          4m56s
+kafka-kafka-1                                   1/1     Running   0          4m28s
+kafka-kafka-2                                   1/1     Running   0          3m51s
+kafka-user-workload-consumer-ffbbc6777-w75wj    1/1     Running   0          4m56s
+kafka-user-workload-producer-85b647f677-7627j   1/1     Running   0          4m56s
+zk-zookeeper-0                                  1/1     Running   0          128m
+zk-zookeeper-1                                  1/1     Running   0          128m
+zk-zookeeper-2                                  1/1     Running   0          128m
+```
+
```

Comment:
can you add here all commands together? 
```
kubectl create ns kudo-kafka
kubectl kudo install zookeeper --instance=zk --namespace=kudo-kafka
kubectl kudo install kafka --instance=kafka --namespace=kudo-kafka \
 -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
 -p RUN_USER_WORKLOAD=true \
 -p ADD_SERVICE_MONITOR=true
```

Reply:
So, does it makes a summary of commands, or just replacing all output code with commands that needs to run and then show the relative output of pods being deployed?

Reply:
summary of the commands, the necessary to have it up and running in a go